tsml_eval/estimators/clustering/ksc/_shift_invariant_distance.py

```diff
@@ -0,0 +1,145 @@
+from typing import List, Optional, Union
+
+import numpy as np
+from numba import njit
+from numba.typed import List as NumbaList
+
+from aeon.distances._utils import _convert_to_list, _is_multivariate
+
+
+@njit(cache=True, fastmath=True)
+def shift_invariant_distance(x: np.ndarray, y: np.ndarray, max_shift: int = 2) -> float:
+    if x.ndim == 1 and y.ndim == 1:
+        dist, _ = _univariate_shift_invariant_distance(x, y, max_shift)
+        return dist
+    if x.ndim == 2 and y.ndim == 2:
+        if x.shape[0] == 1 and y.shape[0] == 1:
+            dist, _ = _univariate_shift_invariant_distance(x[0, :], y[0, :], max_shift)
+            return dist
+        else:
+            n_channels = min(x.shape[0], y.shape[0])
+            distance = 0.0
+            for i in range(n_channels):
+                dist, _ = _univariate_shift_invariant_distance(x[i], y[i], max_shift)
+                distance += dist
+
+            return distance
+
+    raise ValueError("x and y must be 1D or 2D")
+
+
+def shift_invariant_pairwise_distance(
+    X: Union[np.ndarray, List[np.ndarray]],
+    y: Optional[Union[np.ndarray, List[np.ndarray]]] = None,
+    max_shift: int = 2,
+) -> np.ndarray:
+    multivariate_conversion = _is_multivariate(X, y)
+    _X, _ = _convert_to_list(X, "", multivariate_conversion)
+
+    if y is None:
+        return _shift_invariant_pairwise_distance_single(_X, max_shift)
+
+    _y, _ = _convert_to_list(y, "y", multivariate_conversion)
+    return _shift_invariant_pairwise_distance(_X, _y, max_shift)
+
+
+# @njit(cache=True, fastmath=True)
```

Comment:
This is intentional for now as eventually I want this to be numba'd but it was giving me issues.